metadata-ingestion/src/datahub/utilities/source_helpers.py

```diff
@@ -36,7 +36,12 @@ def auto_status_aspect(
     for wu in stream:
         urn = wu.get_urn()
         all_urns.add(urn)
-        if isinstance(wu.metadata, MetadataChangeEventClass):
+
+        if not wu.is_primary_source:
+            # If this is a non-primary source, we pretend like we've seen the status
+            # aspect so that we don't try to emit a removal for it.
+            status_urns.add(urn)
```

Comment:
Ooo this is weird. Might want to rename the  "status_urns" variable to be something more like.... excluded_urns or something